dubbo-registry/dubbo-registry-api/src/test/java/com/alibaba/dubbo/registry/support/AbstractRegistryFactoryTest.java

```diff
@@ -89,7 +89,7 @@ public void testRegistryFactoryCache() throws Exception {
     public void testRegistryFactoryIpCache() throws Exception {
         Registry registry1 = registryFactory.getRegistry(URL.valueOf("dubbo://" + NetUtils.getLocalAddress().getHostName() + ":2233"));
         Registry registry2 = registryFactory.getRegistry(URL.valueOf("dubbo://" + NetUtils.getLocalAddress().getHostAddress() + ":2233"));
-        Assert.assertEquals(registry1, registry2);
+        Assert.assertNotSame(registry1, registry2);

```

Comment:
unit test fails, it should be:

```java
    public void testRegistryFactoryIpCache() throws Exception {
        Registry registry1 = registryFactory.getRegistry(URL.valueOf("dubbo://" + NetUtils.getLocalAddress().getHostName() + ":2233"));
        Registry registry2 = registryFactory.getRegistry(URL.valueOf("dubbo://" + NetUtils.getLocalAddress().getHostAddress() + ":2233"));
        Assertions.assertEquals(registry1, registry2);
    }
```

Reply:
I think registry1 and registry2 should be two different instances, but still don't know why it failed.